neps/nep-0364.md

```diff
@@ -0,0 +1,153 @@
+---
+NEP: 364
+Title: Efficient signature verification and hashing precompile functions
+Author: Blas Rodriguez Irizar <rodrigblas@gmail.com>
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/364
+Status: Draft
+Type: Runtime Spec
+Category: Contract
+Created: 15-Jun-2022
+---
+
+## Summary
+
+This NEP introduces the request of adding into the NEAR runtime a set of pre-compile
+functions that can help IBC compatible light clients run on-chain.
+
+## Motivation
+
+Signature verification and hashing are ubiquitous operations in light clients,
+especially in PoS consensus mechanisms. Our rough numbers say that we’ll be
+verifying ~200 signatures every minute (Polkadot’s authority set is 300 signers).
+
+Therefore, we need a mechanism to perform these operations in a cost-effective way
+in terms of gas and speed. Currently, NEAR does not have any native signature verification toolbox.
+This implies that a light client operating insid NEAR will have to import a library
+compiled to WASM as mentioned in [Zulip](https://near.zulipchat.com/#narrow/stream/295302-general/topic/light_client).
+
+With regards to hashing, there are already several hashing functions implemented natively.
+For our use case, we need to be able to use `blake2b` has function, which is not available.
```

Comment:
```suggestion
For our use case, we need to be able to use the `blake2b` hash function, which is not currently available.
```

Reply:
Removed blake2b from this NEP. Will likely create another one that requests for all the hashing functions that we need.